enhancements/baremetal-IPI/baremetal.md

```diff
@@ -0,0 +1,220 @@
+
+The installer is responsible for kicking off a Baremetal IPI deployment
+with the right configuration.
+
+A set of follow-up enhancement requests will try to bring in the following
+enhancements:
+1. Providing the BMO and it supporting containers with configuration that
+is specific to the server and its provisioning environment. This enhancement
+request is already under review here : 
+https://github.com/openshift/enhancements/pull/90
+
+2. The MAO would also be responsible for deploying a DeamonSet that would
+run on all baremetal servers to monitor the hardware inventory on each
+of these servers.
+
+### User Stories [optional]
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+The MAO already deploys the "metal3" pods with the BareMetal Operator and the
+other (Ironic) containers. Currently, there is no mechanism in place to provide
+these containers with their configuration.
+
+### Risks and Mitigations
+
+Will be specified in follow-up enhancement requests mentioned above.
+
+## Design Details
+
+### Test Plan
+
+True e2e and integration testing can happen only after implementation for
+enhancement https://github.com/openshift/enhancements/pull/90 lands.
+Until then, e2e testing is being performed with the help of some
+developer scripts.
+
+Unit tests have been added to MAO and the Installer to test additions
+made for the Baremetal IPI case.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
```

Comment:
The MAO handles upgrades for us, right? We haven't had any API changes to deal with, yet.